tests/installation/zdup.pm

```diff
@@ -58,9 +58,16 @@ sub run() {
     $out = wait_serial([$zypper_dup_continue, $zypper_dup_conflict, $zypper_dup_error], 240);
     while($out) {
         if ($out =~ $zypper_dup_conflict) {
-            record_soft_failure;
-            send_key '1', 1;
-            send_key 'ret', 1;
+            if (get_var("WORKAROUND_DEPS")) {
+                record_soft_failure;
+                send_key '1', 1;
+                send_key 'ret', 1;
+            }
+            else {
+                $self->result('fail');
+                save_screenshot;
+                return;
```

Comment:
I think this part needs to be redone in the whole zdup. I wasn't entirely familiar with wait_serial when I was producing this.
wait_serial already register screenshots, so save_screenshot is redundant and you can see it in current zdup tests. Also since it immediately returns from run, I think dying will be better suited here (together with mentioning why it died).